app/controlplane/internal/authz/authz_test.go

```diff
@@ -0,0 +1,163 @@
+//
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Authorization package
+package authz
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAddPolicies(t *testing.T) {
+	testcases := []struct {
+		name     string
+		subject  *SubjectAPIToken
+		policies []*Policy
+		wantErr  bool
+	}{
+		{
+			name:    "empty policies",
+			wantErr: true,
+		},
+		{
+			name: "no subject",
+			policies: []*Policy{
+				PolicyWorkflowContractList,
+			},
+			wantErr: true,
+		},
+		{
+			name: "happy path",
+			subject: &SubjectAPIToken{
+				ID: uuid.NewString(),
+			},
+			policies: []*Policy{
+				PolicyWorkflowContractList,
+				PolicyReferrerRead,
+			},
+		},
+		{
+			name: "another happy path",
+			subject: &SubjectAPIToken{
+				ID: uuid.NewString(),
+			},
+			policies: []*Policy{
+				PolicyWorkflowContractList,
+				PolicyWorkflowContractRead,
+				PolicyWorkflowContractUpdate,
+				PolicyArtifactDownload,
+			},
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			enforcer, closer := testEnforcer(t)
+			defer closer.Close()
```

Comment:
Small thing: the inner `t` is shadowing the outer `t`. Maybe it's fine, but I'm just pointing it out, just in case you intended something else.

Small thing n2: you're closing many files at the end of the test function, not at the end of each iteration loop. Even this can be super fine, just pointing it out that you're keeping `len(testcases)` files open, one in its own temp directory, until the end of the test.

Reply:
Thanks, will review this.

I am curious why the linter didn't caught the shadowing though, maybe because it's in a testing file?

Reply:
Ohh yes you're right! I focused on the `for` and I didn't notice there is a `func` there!

Reply:
No worries. Thanks for raising the possibility of mishandling. Usually, I overlook cleanup procedures.